tests/test_utils.py

```diff
@@ -1,65 +1,70 @@
 from datetime import datetime
 from pathlib import Path
-from unittest.mock import call, patch
+from unittest.mock import patch
 
 import pytest
-from botocore.stub import ANY, Stubber  # TODO: fully transition to moto for S3 mocking
 from moto import mock_aws
 from moto.core import patch_client
 
 import opera_disp_tms.utils as ut
 
 
 @pytest.fixture
-def s3_stubber():
-    with Stubber(ut.S3_CLIENT) as stubber:
-        yield stubber
-        stubber.assert_no_pending_responses()
+def s3_bucket():
+    with mock_aws():
+        patch_client(ut.S3_CLIENT)
+
+        bucket_name = 'myBucket'
+        location = {'LocationConstraint': 'us-west-2'}
+
+        ut.S3_CLIENT.create_bucket(Bucket=bucket_name, CreateBucketConfiguration=location)
+
+        yield bucket_name
 
 
 def test_within_in_day():
     assert ut.within_one_day(datetime(2021, 1, 1, 12, 1, 1), datetime(2021, 1, 2, 0, 0, 0))
     assert not ut.within_one_day(datetime(2021, 1, 1, 12, 1, 1), datetime(2021, 1, 2, 12, 1, 2))
 
 
-def test_upload_file_to_s3(tmp_path, s3_stubber):
-    expected_params = {
-        'Body': ANY,
-        'Bucket': 'myBucket',
-        'Key': 'myPrefix/myObject.png',
-        'ContentType': 'image/png',
-        'ChecksumAlgorithm': 'CRC32',
-    }
-    tag_params = {
-        'Bucket': 'myBucket',
-        'Key': 'myPrefix/myObject.png',
-        'Tagging': {'TagSet': [{'Key': 'file_type', 'Value': 'product'}]},
-    }
-    s3_stubber.add_response(method='put_object', expected_params=expected_params, service_response={})
-    s3_stubber.add_response(method='put_object_tagging', expected_params=tag_params, service_response={})
-
+@mock_aws
+def test_upload_file_to_s3(tmp_path, s3_bucket):
     file_to_upload = tmp_path / 'myFile.png'
     file_to_upload.touch()
-    ut.upload_file_to_s3(file_to_upload, 'myBucket', key='myPrefix/myObject.png')
 
+    ut.upload_file_to_s3(file_to_upload, s3_bucket, key='myPrefix/myObject.png')
+
+    resp = ut.S3_CLIENT.get_object(Bucket=s3_bucket, Key='myPrefix/myObject.png')
+
+    assert resp['ContentType'] == 'image/png'
+
+    tag_resp = ut.S3_CLIENT.get_object_tagging(Bucket=s3_bucket, Key='myPrefix/myObject.png')
+
+    assert len(tag_resp['TagSet']) == 1
+    tag = tag_resp['TagSet'].pop()
+
+    assert tag['Key'] == 'file_type'
+    assert tag['Value'] == 'product'
```

Comment:
I prefer to assert the entire object structure where possible:
```suggestion
    assert tag_resp['TagSet'] == [{'Key': 'file_type', 'Value': 'product'}]
```